src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,274 @@
+                            task = new Deadline(desc, DateTimeParser.deadlineDateParse(date));
+                        }
+                    }
+                    break;
+                case "event":
+                    if (hasEmptyDesc(splitTasks)) {
+                        throw new EmptyDescriptionException(
+                                "Sorry the description of an event cannot be empty" + Constants.EVENT_FORMAT
+                        );
+                    } else {
+                        String[] parsedEvent = splitTasks[1].split("/at");
+                        if (hasDateButEmptyDesc(parsedEvent)) {
+                            throw new EmptyDescriptionException(
+                                    "Sorry the description of an event cannot be empty" + Constants.EVENT_FORMAT
+                            );
+                        } else if (hasEmptyDesc(parsedEvent)) {
+                            throw new IncorrectFormatException(
+                                    "Please add a date and time for your event!" + Constants.EVENT_FORMAT);
+                        } else {
+                            String details = parsedEvent[0].trim();
+                            String at = parsedEvent[1].trim();
+                            task = new Event(details, DateTimeParser.eventDateTimeParse(at));
+                        }
+                    }
+                    break;
+                default:
+                    throw new InvalidKeywordException();
+                }
+            return new AddCommand(task);
+        } catch (ArrayIndexOutOfBoundsException ex) {
+                switch (taskType) {
+                case "deadline": throw new IncorrectFormatException(
+                        "Please specify a description and date for your deadline!" + Constants.DEADLINE_FORMAT);
+                case "event": throw new IncorrectFormatException(
+                        "Please specify a description, date and time for your event!" + Constants.EVENT_FORMAT);
+                default: throw new IncorrectFormatException(
+                        "Please specify a description and date/time for your task!");
+                }
+        }
+    }
+
+    /**
+     * The parseDone method is a task-specific parse function which is called when the user uses the
+     * 'done' keyword
+     *
+     * @param input the user's string input
+     * @param taskList the list to be operated on
+     * @return a DoneCommand to execute the done operation
+     */
+    private static DoneCommand parseDone(String input, TaskList taskList) {
+        try {
+            String[] parsedTask = input.split("\\s", 2);
+            String indexOfTask = parsedTask[1].trim();
+            int index = Integer.parseInt(indexOfTask);
+            if (index > 0 && index > taskList.size()) {
+                throw new NonExistentTaskException();
+            } else {
+                return new DoneCommand(index);
+            }
```

Comment:
The `else` may be omitted.
```suggestion
            }
            
            return new DoneCommand(index);
```